server/plugin/virtual_agent.go

```diff
@@ -329,13 +328,17 @@ func (p *Plugin) ProcessResponse(data []byte) error {
 				}
 			}
 		case *OutputImage:
-			var post *model.Post
-			post, err = p.CreateOutputImagePost(res, userID)
-			if err != nil {
-				return err
+			linkContents := strings.Split(res.Value, "/")
+			if len(linkContents) < 1 {
+				if _, err = p.DM(userID, res.AltText); err != nil {
+					return err
+				}
+				p.API.LogError(InvalidImageLinkError)
+				return errors.New(InvalidImageLinkError)
 			}
 
-			if _, err = p.dm(userID, post); err != nil {
+			completeFilename := linkContents[len(linkContents)-1]
+			if _, err = p.DM(userID, fmt.Sprintf("![%s](%s)", completeFilename, res.Value)); err != nil {
```

Comment:
what is the use of this `!`?

Reply:
It renders the image present in the link. Just like we have on github